platform/src/main/java/org/hillview/utils/IntSet.java

```diff
@@ -97,6 +97,19 @@ public boolean add(final int k) {
         return true;
     }
 
+    /**
+     *
+     * @param pos a location in the key[] array
+     * @return in res[0] the location of the next full slot after cursor. In res[1] the value.
+     */
+    public int[] getNext(int pos) {
+        int[] res = new int[2];
```

Comment:
This is expensive: you allocate an array for every call.
It's better to pass the array as an out argument, or to expose the internal indexing operation and return just the internal index as an opaque value.

Reply:
You are right, I made the change